src/pushsource/_impl/source.py

```diff
@@ -63,7 +63,7 @@ def __iter__(self):
             if not hasattr(item, "src") or not item.src or not item.src.startswith("/"):
                 yield item
             else:
-                wait_exist(item.src, timeout, poll_rate)
+                wait_exist(item, timeout, poll_rate)
```

Comment:
I think this change is probably not needed, or we should find some way to limit when this happens.

The logic here was introduced purely to support a replication problem between some specific NFS mounts.  I believe it might be obsolete now, but even if not - this is NOT something I think we should support for anything other than this known NFS case.

I don't think this particular bit of code needs to support different ways of reading the item, I think it is reasonable to take this position:

- if `content` is later customized to do something other than just open the file: we should try hard to avoid a situation where the new method of opening would for some reason need to poll for the content's existence
- if we really can't avoid that situation: then it is within the scope of the `content` implementation to do the polling. For instance if there is an implementation of `content` which fetches content via HTTP requests, it could loop a few times in case of 404 errors, if and only if it's known to make sense in that particular case.

Overall, I would suggest leaving this code alone for now.  It's not necessary to make this change to satisfy the main goal of this issue.  If `wait_exist` is still needed at all, it's only needed for this specific NFS replication issue, but it's not obvious to me how to make the code only be applied in that case.

I think it'd be better to revisit it when it actually comes time to introduce a different implementation of `content`, and with any luck `wait_exist` could be completely removed by then.